src/cqrs/ExecutionContext.ts

```diff
@@ -139,19 +140,29 @@ export class ExecutionContext extends AggregateEventStore {
         await commandExecutor.execute(command, this, aggregate);
       } catch (e) {
         this.committed = true;
-        this.error = new Error(`There was an error executing command ${command}`);
-        this.error['cause'] = e;
-        throw this.error;
+        if (e instanceof ReturnToCallerError) {
+          throw e;
+        } else {
+          // hide the actual error and return a custom error
+          this.error = new ReturnToCallerError(`There was an error executing command ${command}`);
```

Comment:
Why force all errors into a ReturnToCallerError? If you do this all errors will be shown to the client?

Reply:
I mean... I understand why if it is a ReturnToCallerError you just re-throw it. That's fine... But if it's not, you shouldn't convert it to a ReturnToCallerError.